Hide useless asteroid warnings when running tests

mbertrand opened the pull request:
#### What are the relevant tickets?
Closes #1349

#### What's this PR do?
Prevents asteroid warnings (deprecations, etc) from showing up after running tests locally.

#### How should this be manually tested?
Run `docker-compose run web tox`.  There should not be a huge stream of asteroid warnings afterward.

noisecapella commented:
It seems like this prevents pylint from running... I think we need to revert the change in the other repo that @amir-qayyum-khan made

noisecapella commented:
I think the trailing comma is what's preventing pylint from issuing any errors. But if I remove the trailing comma and leave 'astroid' I see all the usual warnings and errors like I do on master

mbertrand commented:
Sorry about that, I thought pylint was responsible for complaining about pep8 violations so assumed everything was still working when I added some extra lines in a file and got an error about it.

Took a different approach and updated astroid, pylint, etc to latest versions.  This got rid of the astroid errors but also introduced some new lint violations.  I added frequent ones to the disable list in `pylintrc`, then added `disable` lines where appropriate for the rest.

mbertrand commented on pylintrc:
```diff
@@ -17,6 +17,7 @@ ignored-classes=
 
 [MESSAGES CONTROL]
-disable = no-member, old-style-class, no-init, too-few-public-methods, abstract-method, invalid-name, too-many-ancestors, line-too-long, no-self-use, len-as-condition, no-else-return
+disable = no-member, old-style-class, no-init, too-few-public-methods, abstract-method, invalid-name, too-many-ancestors, line-too-long, no-self-use, len-as-condition, no-else-return, cyclic-import, duplicate-code, inconsistent-return-statements, keyword-arg-before-vararg, consider-using-set-comprehension, inconsistent-return-statements, raising-format-tuple
```
Removed keyword-arg-before-vararg, consider-using-set-comprehension, raising-format-tuple from the ignore list

noisecapella commented on authentication/pipeline/user.py:
```diff
@@ -20,7 +20,7 @@
 
 
-def validate_email_auth_request(strategy, backend, user=None, *args, **kwargs):  # pylint: disable=unused-argument
+def validate_email_auth_request(strategy, backend, user=None, **kwargs):  # pylint: disable=unused-argument
```
We should still keep these around just in case there is an extra positional argument and we don't want to error on that. But we can move `*args` before `user=None` to fix the warning

mbertrand replied:
Moving `*args` before `user` causes a test to fail (`auth.pipeline.user_test.test_get_username`).  I think it might be best to just leave these functions in their original state and add an ignore for `keyword-arg-before-vararg` to the top, so as to avoid any nasty surprises with social-auth.

mbertrand replied:
I reverted the code for `auth.pipeline.user` to be the same as master and added `# pylint: disable=keyword-arg-before-vararg` to the top of the file, I think this is safer for now.

noisecapella commented on channels/serializers.py:
```diff
@@ -900,4 +900,4 @@ def get_reasons(self, instance):
             list of str: list of reasons a post/comment has been reported for
         """
-        return sorted(list(set([report[0] for report in instance.user_reports + instance.mod_reports])))
+        return sorted(list({report[0] for report in instance.user_reports + instance.mod_reports}))
```
You can also remove the `list` at the beginning since `sorted` will accept a set as input

mbertrand replied:
👍
